### ISSUE TITLE
The issue about image preproccess and prediciton

### ISSUE THREAD

**Issue by xuhuimi** [satyenrajpal/Semantic-Segmentation-U-Net#1]

**Comment by xuhuimi** [comment]
Hello! I'm really appreciate that you share the code with us. Now I'm doing my graduation thesis about semantic segmentation with UNet. When I run you code with vaihingen images, I got "Cross Entropy Loss: NAN" and "class-wise accuracy:[1,0,0,0,0]" and the IoU fluctuated around 0.26. I don't know why. I try some image preproccess methods but still got the bad result. So do you make some othe preproccess methods?
What's more, I search many information about the image prediction, but I can't figure out. Can you share the predicition code with me?
This expriment is really important for me! My email is 578270791@qq.com. Look forward to your reply. Thank you very much!

**Comment by xuhuimi** [comment]
I made a keras implementation of Semantic Segmentation based on UNet, but got unsatisfactory results. I adjusted the parameter many days and tried other methods to make the results better, but all failed. I have to finised my expriment in due time or  I can't graduate on time. So I really hope you can help me. Thank you!

**Comment by satyenrajpal** [comment]
This network breaks down an image into 572x572 sized images and then passes it through the network. The reason being that Vaihingen images are huge and difficult to fit into memory. You might want to check that. No, I didn't do any pre-processing. I didn't get what you meant by image prediction code. Can you please elaborate.
What exactly do you mean by unsatisfactory results?

**Comment by xuhuimi** [comment]
Thank you for your reply!
1.When I run your code, I do break down the image into 572*572 size as you said, but I still got the result"Cross Entropy Loss: NAN" and "class-wise accuracy:[1,0,0,0,0]" and the IoU fluctuated around 0.26. I thought it was the problem of not doing image pre-processing like image normalization, the label value from 0 to 255 transform to [0,1,2,3,4,5](cause I saw some networks make this kind of transformation), etc. Because I found your format of image is ".jpg" and label is ".png", not the same with the original data downloaded from ISPRS. 
2.The image prediction means how to load the model. I'm not good at tensorflow architecture, so I don't know how to get the output image using the model.
3. I have done some experiments using my own code with keras architecture. My loss value is always around 103 and the accuray is less than 0.3. The network doesn't converge! So it's obviously that the output images has a big gap with the labels. So I got unsatisfactory results with my own code. Here are some experiment results I got.






------------------ 原始邮件 ------------------
发件人: "Satyen Rajpal"<notifications@github.com>;
发送时间: 2018年3月23日(星期五) 中午11:25
收件人: "satyenrajpal/Semantic-Segmentation-U-Net"<Semantic-Segmentation-U-Net@noreply.github.com>;
抄送: "西南交通大学许慧敏"<578270791@qq.com>; "State change"<state_change@noreply.github.com>; 
主题: Re: [satyenrajpal/Semantic-Segmentation-U-Net] The issue about imagepreproccess and prediciton (#1)




This network breaks down an image into 572x572 sized images and then passes it through the network. The reason being that Vaihingen images are huge and difficult to fit into memory. You might want to check that. No, I didn't do any pre-processing. I didn't get what you meant by image prediction code. Can you please elaborate.
 What exactly do you mean by unsatisfactory results?

**Comment by xuhuimi** [comment]
Tensorflow is really difficult to understand for me :( . So can you show me how to get the output image using the model with one original RGB image. And I also don't know why I  got the result "Cross Entropy Loss:NAN" when I run your code. Look forward to your reply.

**Comment by satyenrajpal** [comment]
How are your class numbers allocated? This code assumes that the classes are from 1 to 6 (a total of 6 classes). For obtaining the predicted image, I'll upload a code shortly.
Learning TensorFlow is an arduous journey but a fulfilling one. There are a ton of tutorials available online which you could look into. Maybe start with basic classification problems with easily available datasets such as MNIST and then work your way into segmentation. You could also look at pyTorch. It's better suited for research and might be better in your case.

**Comment by xuhuimi** [comment]
I'm really appreciate with your help! 
I predict the vaihingen images for 6 classes. When I run the code, the class numbers I alllcated is 0 to 5.(Maybe that's the promblem. I will test it.)
I'm better at keras framwork.The tensorflow is much harder for me to understand. I will learn some tutorials about tensorflow. Thank you for your advice!
